iib/web/migrations/versions/04dd7532d9c5_polymorphic_requests.py

```diff
@@ -0,0 +1,382 @@
+        )
+    )
+
+    connection.execute(
+        request_rm_table.insert().from_select(
+            [
+                'id',
+                'binary_image_id',
+                'binary_image_resolved_id',
+                'from_index_id',
+                'from_index_resolved_id',
+                'index_image_id',
+            ],
+            select(
+                [
+                    old_request_table.c.id,
+                    old_request_table.c.binary_image_id,
+                    old_request_table.c.binary_image_resolved_id,
+                    old_request_table.c.from_index_id,
+                    old_request_table.c.from_index_resolved_id,
+                    old_request_table.c.index_image_id,
+                ],
+                old_request_table.c.type == 2,
+            ),
+        )
+    )
+
+
+def downgrade():
+    with op.batch_alter_table('request') as batch_op:
+        batch_op.add_column(sa.Column('index_image_id', sa.INTEGER(), nullable=True))
+        batch_op.add_column(sa.Column('organization', sa.VARCHAR(), nullable=True))
+        batch_op.add_column(sa.Column('from_index_resolved_id', sa.INTEGER(), nullable=True))
+        # We need to make this nullable, migrate the data, then we can set it
+        # to not nullable
+        batch_op.add_column(sa.Column('binary_image_id', sa.INTEGER(), nullable=True))
+        batch_op.add_column(sa.Column('from_index_id', sa.INTEGER(), nullable=True))
+        batch_op.add_column(sa.Column('binary_image_resolved_id', sa.INTEGER(), nullable=True))
+        batch_op.create_foreign_key(
+            'request_from_index_resolved_id_fkey', 'image', ['from_index_resolved_id'], ['id'],
+        )
+        batch_op.create_foreign_key(
+            'request_index_image_id_fkey', 'image', ['index_image_id'], ['id'],
+        )
+        batch_op.create_foreign_key(
+            'request_binary_image_resolved_id_fkey', 'image', ['binary_image_resolved_id'], ['id'],
+        )
+        batch_op.create_foreign_key(
+            'request_binary_image_id_fkey', 'image', ['binary_image_id'], ['id'],
+        )
+        batch_op.create_foreign_key(
+            'request_from_index_id_fkey', 'image', ['from_index_id'], ['id'],
+        )
+
+    op.create_table(
+        'request_operator',
+        sa.Column('request_id', sa.INTEGER(), nullable=False),
+        sa.Column('operator_id', sa.INTEGER(), nullable=False),
+        sa.ForeignKeyConstraint(['operator_id'], ['operator.id']),
+        sa.ForeignKeyConstraint(['request_id'], ['request.id']),
+        sa.PrimaryKeyConstraint('request_id', 'operator_id'),
+        sa.UniqueConstraint('request_id', 'operator_id'),
+    )
+    with op.batch_alter_table('request_operator') as batch_op:
+        batch_op.create_index('ix_request_operator_request_id', ['request_id'], unique=False)
+        batch_op.create_index('ix_request_operator_operator_id', ['operator_id'], unique=False)
```

Comment:
As far as I know, a batch operation is not needed here:
```suggestion
    op.create_index('ix_request_operator_request_id', 'request_operator', ['request_id'], unique=False)
    op.create_index('ix_request_operator_operator_id', 'request_operator', ['operator_id'], unique=False)
```